File: test.spritebuilder/Source/TDEngine/Map/Detachment.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package Map;

import internalObject.SolderInternal;

/**
 *
 * @author Константин
 */
 public final  class  Detachment 
    {  
     public final int BIG_COUNT = 30;
     public final int MEDIUM_COUNT = 20;
     public final int SMALL_COUNT = 10;   
     public int getCount()
     {
             if(size == SizeGroupe.small) return SMALL_COUNT;
             if(size == SizeGroupe.medium) return MEDIUM_COUNT;
             if(size == SizeGroupe.big) return BIG_COUNT; 
             return 0;
     }
        public enum SizeGroupe{small,medium,big}  
        public  SizeGroupe size; 
        public SolderInternal.SolderType type;
       
       
 }

File: TowerDefence.spritebuilder/Map/Detachment.swift
//
//  Detachment.swift
//  TowerDefence
//
//  Created by koveres on 05.12.14.
//  Copyright (c) 2014 Apportable. All rights reserved.
//

import Foundation


public enum SizeGroupe{case small,medium,big}


public   class  Detachment
{
    public var  BIG_COUNT:CInt = 30;
    public var   MEDIUM_COUNT:CInt = 20;
    public var   SMALL_COUNT:CInt = 10;
    public func getCount() ->(CInt)
    {
        if(size == SizeGroupe.small){ return SMALL_COUNT;}
        if(size == SizeGroupe.medium){ return MEDIUM_COUNT;}
        if(size == SizeGroupe.big){ return BIG_COUNT;}
        return 0;
    }
    public init(){}
    public  var size:SizeGroupe?;
    public var  type:SolderType?;
    
    
}